source/common/stats/thread_local_store.cc

```diff
@@ -193,9 +193,17 @@ StatType& ThreadLocalStoreImpl::ScopeImpl::safeMakeStat(
     const std::string& name, StatMap<std::shared_ptr<StatType>>& central_cache_map,
     MakeStatFn<StatType> make_stat, StatMap<std::shared_ptr<StatType>>* tls_cache) {
 
+  const char* stat_key = name.c_str();
+  std::unique_ptr<std::string> truncation_buffer;
+  absl::string_view truncated_name = parent_.truncateStatNameIfNeeded(name);
```

Comment:
This will print the warning on every call, right? Thoughts on rejiggering so that we only print the warning when the stat is initially created? This could warn at very high rate.

Reply:
good catch. should be easy to fix.

Reply:
test for this fails as expected:
```
TEST_F(TruncatingAllocTest, CounterTruncated) {
  Counter* counter = nullptr;
  EXPECT_LOG_CONTAINS("warning", "is too long with", {
      Counter& c = store_->counter(long_name_);
      counter = &c;
    });
  EXPECT_NO_LOGS(EXPECT_EQ(counter, &store_->counter(long_name_)));
}

TEST_F(TruncatingAllocTest, GaugeTruncated) {
  Gauge* gauge = nullptr;
  EXPECT_LOG_CONTAINS("warning", "is too long with", {
      Gauge& g = store_->gauge(long_name_);
      gauge = &g;
    });
  EXPECT_NO_LOGS(EXPECT_EQ(gauge, &store_->gauge(long_name_)));
}
```